deployment-bundle-steps.md

```diff
@@ -0,0 +1,135 @@
+# How to setup a project with deployment to S3
+
+1. Create S3 bucket for production / staging (your-bucket-name-production & your-bucket-name-staging). (If using cloudfront, see [optional settings](#optional-settings))
+
+2. Log into [circleci.com](https://circleci.com/) and add your project. Set up the following credentials: (see other members of the team for these values).
+  - Environment Variables
+    - GHUSER
+    - GHPASS
+    - GRAPHITE_API_KEY
+    - GRAPHITE_ENDPOINT_PREFIX
+    - CLOUDFRONT_DISTRIBUTION_ID (if using cloudfront for your production bundle)
+  - AWS CodeDeploy
+    - Access Key Id
+    - Secret Access Key
+
+3. Create relevant files for your project:
+  - [circle.yml](#circleyml)
+  - [scripts/deployS3.sh](#deploys3sh)
+  - [istanbul.yml](#istanbulyml)
+
+4. Add [scripts](#npm-scripts) to your package.json
+
+5. Add [devDependencies](#devdependencies) to your package.json
+
+6. Make sure your tests & linting pass. Merge to staging, see it deploy!
+
+---
+
+##### Optional Settings
+
+1a. (optional) Setup cloud front on the production bucket & add the following to the end of your `deployS3.sh` script:
+```bash
+if [ $1 == 'production' ]; then
+  aws cloudfront create-invalidation --distribution-id $CLOUDFRONT_DISTRIBUTION_ID --paths /\*
```

Comment:
yep in practice we are doing it as @robhuzzey describes (I know tracker does and the last booking engines I worked on did too) but we probably should be doing it as @shackpank describes for better cache management. For an ultimate-shared-tech-document maybe we should cover both cases or neither. we could upload as both `engine.min.js` and `engine.v3.2.1.min.js` and only bust the cache on `engine.min.js` maybe?